fmt/src/selection.rs

```diff
@@ -182,15 +180,45 @@ fn select_files_with_git(
         builder.build().context(SelectWithIgnoreSnafu)?
     };
 
-    let mut it = WalkDir::new(basedir).into_iter();
+    let basedir = basedir
```

Comment:
Canonicalizing the basedir means that all paths will be canonical, which means workdir and iterated paths will naturally be comparable.

Reply:
Good point.